TestHelper.psm1

```diff
@@ -73,3 +79,159 @@ function New-Nuspec
+                -f $Script:DesignerModuleName))
+        {
+            Import-Module PackageManagement
+
+            # Make sure the Nuget Package provider is initialized.
+            $null = Get-PackageProvider -name nuget -ForceBootStrap -Force
+
+            # PowerShellGet is available - use that
+            Import-Module PowerShellGet
+
+            # Install the module
+            Install-Module -Name $Script:DesignerModuleName -Force
+
+            Write-Verbose -Verbose (`
+                'The {0} module was installed using PowerShellGet.' `
+                    -f $Script:DesignerModuleName            
+            )
+        }
+        else
+        {
+            Write-Warning -Message (`
+                '{0} module was not installed automatically.' `
+                    -f $Script:DesignerModuleName
+            )
+            return $null
+        }
+    }
+    else
+    {
+        # PowerShellGet module isn't available, so use Nuget directly to download it
+        $nugetPath = 'nuget.exe'
+
+        # Can't assume nuget.exe is available - look for it in Path
+        if ((Get-Command $nugetPath -ErrorAction SilentlyContinue) -eq $null) 
+        {
+            # Is it in temp folder?
+            $nugetPath = Join-Path -Path $ENV:Temp -ChildPath $nugetPath
+            if (-not (Test-Path -Path $nugetPath))
+            {
+                # Nuget.exe can't be found - download it to temp folder
+                $NugetDownloadURL = 'http://nuget.org/nuget.exe'
+                If ($Force -or $PSCmdlet.ShouldProcess( `
+                    "Download Nuget.exe from '{0}' to Temp folder" `
+                        -f $NugetDownloadURL))
+                {
+                    Invoke-WebRequest $Script:NugetDownloadURL -OutFile $nugetPath
+
+                    Write-Verbose -Verbose (`
+                        "Nuget.exe was installed from '{0}' to Temp folder." `
+                            -f $Script:NugetDownloadURL
+                    )
+                }
+                else
+                {
+                    # Without Nuget.exe we can't continue
+                    Write-Warning -Message (`
+                        'Nuget.exe was not installed. {0} module can not be installed automatically.' `
+                            -f $Script:DesignerModuleName
+                    )
+                    return $null    
+                }
+            }
+            else
+            {
+                Write-Verbose -Verbose 'Using Nuget.exe found in Temp folder.'
+            }
+        }
+        
+        $nugetSource = 'https://www.powershellgallery.com/api/v2'
+        If ($Force -or $PSCmdlet.ShouldProcess(( `
+            "Download and install the {0} module from '{1}' using Nuget" `
+                -f $Script:DesignerModuleName,$nugetSource)))
+        {
+            # Use Nuget.exe to install the module
+            & "$nugetPath" @( `
```

Comment:
It does have a 100% success rate in all my tests (both in local testing and AppVeyor). So yes, we could definitely do away with the PowerShellGet method and always use Nuget.

I think there was some suggestion and some point to support both methods, but we can definitely keep it simple and get rid of that code path.

Consider it done!